### PR TITLE
Simplify trimmed_mean

### PR DESCRIPTION
**Pull request by vmihailenco**
This is one of the ways to fix https://github.com/CamDavidsonPilon/tdigest/issues/45. I know that this is much simpler but I hope it is not much less precise. I will add test from the https://github.com/CamDavidsonPilon/tdigest/issues/45 if this is acceptable.

### CODE REVIEW

**Comment by CamDavidsonPilon** [comment]
:wave: hi @vmihailenco, thanks for the PR. 

I think adding some tests (for example, duplicating the example in your issue) would be great! Plus maybe some other tests / edge cases you see. 

You can add them to `tdigest/tests/test_tdigest.py` and test with `py.test`

File: tdigest/tdigest.py [hunk 1]
```diff
@@ -222,43 +222,32 @@ def trimmed_mean(self, p1, p2):
...
-
-            if p2 < t + k_i:
-                nu = self.__interpolate(i,key,p2-t)
-                s -= nu * k_i * c_i.mean
-                k -= nu * k_i
-                break
+        min_count = p1 / 100. * self.n
+        max_count = p2 / 100. * self.n
+
+        trimmed_sum = trimmed_count = curr_count = 0
```
Comment (CamDavidsonPilon) [inline tdigest/tdigest.py]: this algorithm seems very reasonable 👍

**Comment by CamDavidsonPilon** [comment]
Overall, PR looks good to me. It's clear to me that I need to add more tests to this function, however. 

Thanks @vmihailenco!
